### PR TITLE
Initial version of a Vert.x Sql Client detector

### PR DESCRIPTION
**Pull request by pmlopes**
Signed-off-by: Paulo Lopes <pmlopes@gmail.com>

This is a initial pull request to add a sql injection detector for vert.x sql client. The code passes the tests but the build is failing due to missing headers on the mock API.

I am wondering how could one solve the issue, as it seems that the remaining mocks do not have any headers, so I'm a bit lost here how to fix it.

### CODE REVIEW

File: findsecbugs-samples-java/src/test/java/testcode/sqli/VertxSqlClient.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package testcode.sqli;
+
+import io.vertx.sqlclient.SqlClient;
+import io.vertx.sqlclient.SqlConnection;
+
+public class VertxSqlClient {
+
+    public void injection1(SqlClient client, String injection) {
+        client.query(injection);
```
Comment (h3xstream) [inline findsecbugs-samples-java/src/test/java/testcode/sqli/VertxSqlClient.java:9]: Example are minimalist but I think the vulnerability type for these API is well known.

File: findsecbugs-plugin/src/main/resources/metadata/messages.xml [hunk 2]
```diff
@@ -2507,6 +2507,60 @@ DB.withConnection { implicit c =>
...
+    .onFailure(err -> ...);
+</pre>
+
+    <b>Solution (using OWASP Encoder):</b><br/>
+
+    <pre>
+import org.owasp.esapi.Encoder;
+
+SqlClient
+    .query( "select * from Customer where id=" + Encoder.encodeForSQL(inputId) )
```
Comment (h3xstream) [inline findsecbugs-plugin/src/main/resources/metadata/messages.xml:2544]: While FSB eliminates false positive when encoding functions are use, they do not represent the best solution.

For instance, in the code given - if `inputId` is a string - a malicious inputId could be `123 union select * from Users`. The string would not be sanitized because the encoding function assume that the value is placed between single quotes or double quotes. A `union` statement or any form of sub-queries could be introduced as long as no single quote is used. (Ref: [MysqlCodec from ESAPI](https://github.com/ESAPI/esapi-java-legacy/blob/d7bb3a537a76954242db94a6dd2cdb4a88a2856c/src/main/java/org/owasp/esapi/codecs/MySQLCodec.java#L177-L188))
```java
SqlClient
    .query( "select * from Customer where id=" + Encoder.encodeForSQL(inputId) )
```

Bottom line, I would simply remove this second solution. The rest is perfect.

File: findsecbugs-samples-java/src/test/java/testcode/sqli/VertxSqlClient.java (hunk 1 above)
Comment (pmlopes) [reply findsecbugs-samples-java/src/test/java/testcode/sqli/VertxSqlClient.java:9]: Yes the vert.x API is fluent so there's just 1 way to pass a sql statement to be executed, there are 2 classes where this happens:

* SqlClient - means a connection managed client where users run queries ad-hoc without worring to get a connection and returning it to the pool.
* SqlConnection - means that the user is being responsible to control the connection and return it to the pool.
